examples/inference/gpt/utils.py

```diff
@@ -313,10 +322,20 @@ def get_requests_from_file(
         len(prompts),
     )
 
+    # Update sampling params with per-prompt values.
+    sampling_params_list = [sampling_params] * len(prompts)
+    if args.num_tokens_from_file:
+        if sampling_params is None:
+            sampling_params = get_default_sampling_params(tokenizer.eod)
+        for i in range(len(prompts)):
+            sp = copy.deepcopy(sampling_params)
+            sp.num_tokens_to_generate = num_tokens_to_generate[i]
+            sampling_params_list[i] = sp
```

Comment:
Why not do this in the file-processing loop above to avoid having to iterate through `prompts` twice?

Reply:
Done!